File: qwencoder-eval/chat/eval-dev-quality/testdata/java/light/src/main/java/com/eval/BalancedBrackets.java
// Copyright (C) rosettacode.org
package com.eval;

public class BalancedBrackets {
  static boolean hasBalancedBrackets(char[] charArray) {
    int brackets = 0;
    for (char ch : charArray) {
      if (ch == '[') {
        brackets++;
      } else if (ch == ']') {
        brackets--;
      } else {
        return false; // Non-bracket characters.
      }
      if (brackets < 0) { // Closing bracket before opening bracket.
        return false;
      }
    }
    return brackets == 0;
  }
}

File: qwencoder-eval/instruct/eval-dev-quality/testdata/java/transpile/balancedBrackets/implementation/balancedBrackets.go
package balancedBrackets

func hasBalancedBrackets(charArray string) bool {
	brackets := 0
	for _, ch := range charArray {
		if ch == '[' {
			brackets++
		} else if ch == ']' {
			brackets--
		} else {
			return false // Non-bracket characters.
		}
	}

	if brackets < 0 { // Closing bracket before opening bracket.
		return false
	}

	return brackets == 0
}
